src/pvnet_app/forecaster.py

```diff
@@ -280,6 +289,19 @@ def log_forecast_to_database(self, session: Session) -> None:
         )
 
         # save to new dataplatform
-        # TODO
-
+        try:
+            asyncio.run(self.save_forecast_to_data_platform())
+        except Exception as e:
+            self.logger.error(f"Failed to save forecast to data platform with error {e}")
+
+    async def save_forecast_to_data_platform(self) -> None:
+        """Save forecast to data platform."""
+        async with Channel(host=data_platform_host, port=data_platform_port) as channel:
+            client = dp.DataPlatformDataServiceStub(channel)
+            await save_forecast_to_data_platform(
+                forecast_da=self.da_abs_all,
+                model_tag=self.model_tag,
+                init_time_utc=self.t0.to_pydatetime().replace(tzinfo=UTC),
+                client=client,
+            )
```

Comment:
Why wrap this in this manner? Couldn't you just call `save_forecast_to_data_platform` with a client made in the try block above? Seems unnecessary

Reply:
I had to make the channel with `async with` which was the motivation for moving it to a seperate function. I could move it back to `log_forecast_to_database` and make that async